src/index.js

```diff
@@ -239,7 +239,9 @@ var enqueuePostPromiseJob =
         process.nextTick(fn);
       });
     } :
-    setImmediate || setTimeout;
+    typeof setImmediate === 'function' ?
+      fn => { setImmediate(fn); } :
```

Comment:
IE 11 throws "Invalid calling object" error if the receiver of `setImmediate` is not a `window`.

```
var o = { m: setImmediate }; o.m(function () { console.log('ok'); });
// => "Invalid calling object" error in IE 11